custom_model_runner/datarobot_drum/drum/gpu_predictors/base.py

```diff
@@ -110,14 +108,56 @@ def is_sidecar(self):
         # When true, DRUM is expected to run as a sidecar, proxying/monitoring requests to a model container.
         return self._params.get("sidecar", False)
 
+    @cached_property
+    def served_model_name(self):
+        """
+        There are two ways to specify the model name:
+        - Explicitly set in the request: Required by the /chat/completions API.
+        - Not set in the request: In this case, we ensure backward compatibility with the /predictions API.
+
+        Next, there are two ways to run the model:
+        - Old way: Single-container deployment, where both DRUM and the OpenAI server run in the same container.
+        - New way: Multi-container deployment, where DRUM runs as a sidecar alongside the OpenAI server.
+
+        In the new approach, the OpenAI server starts separately. Therefore, we must either: set NIM_SERVED_MODEL_NAME
+        in the second container, or retrieve the model name from the API.
+        """
+        served_model_name = self.get_optional_parameter(
+            "served_model_name", self.DEFAULT_MODEL_NAME
+        )
+
+        if not self.is_sidecar:
+            return served_model_name
+
+        model_ids = self._get_deployed_model_ids()
+
+        if not model_ids:
+            # Some containers do not expose the model names API
+            return served_model_name
+        if len(model_ids) == 1:
+            return model_ids[0]
+
+        # Multiple models detected, but only one is expected.
+        raise DrumCommonException(
+            f"Multiple models detected in the NIM container, but only one is expected: {model_ids}."
+        )
```

Comment:
I feel like this is a little risky as we haven't dug into all 62 models yet? Also, it is definitely not forward proof (with LoRA).

I say why not just pick the first one and if there are others in the list oh well? Combine this with my other suggestion and this should still solve the base case but also allow end users a path to utilize those additional models by specifying them explicitly?

Reply:
Changed code to pick the first one and do not crash.